helpers.go

```diff
@@ -310,7 +310,10 @@ func (ctx *e2econtext) waitForOperatorToBeReady(t *testing.T) {
 		if len(od.Status.Conditions) == 0 {
 			return fmt.Errorf("no conditions for deployment yet")
 		}
-		if od.Status.Conditions[0].Type != appsv1.DeploymentAvailable {
+		for cond := range od.Status.Conditions {
+			if cond.Type == appsv1.DeploymentAvailable {
+				return nil
+			}
 			return fmt.Errorf("the deployment is not ready yet")
```

Comment:
We can probably fix this in another patch, but in the case where the operator fails to come up do we want to proceed with the test?

Right now, the suite runs all the subsequent tests even if operator deployments fail to come up because we're not checking the error condition on the caller.

https://github.com/ComplianceAsCode/ocp4e2e/blob/main/e2e_test.go#L25

Reply:
I can do add the check for the error condition.
I noticed that and forgot to implement it.